### ISSUE TITLE
Invalid parse of arrow functions in array declaration without indexes

### ISSUE THREAD

**Issue by KartaviK** [squizlabs/PHP_CodeSniffer#3095]
**Describe the bug**
Invalid parsing of code for array with arrow functions

**Code sample**
```php
$funcs = [
    static fn () => null,
    static fn () => null,
];
```

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php ...`
3. See error message displayed
```
FOUND 2 ERRORS AFFECTING 2 LINES
----------------------------------------------------------------------
xx | ERROR | [x] The first value in a multi-value array must be on a
    |       |     new line
xx | ERROR | [x] Each value in a multi-line array must be on a new
    |       |     line
----------------------------------------------------------------------
PHPCBF CAN FIX THE 2 MARKED SNIFF VIOLATIONS AUTOMATICALLY
```

**Expected behavior**
Not errors found

**Versions (please complete the following information):**
 - OS: Ubuntu
 - PHP: 7.4
 - PHPCS: 3.5.6
 - Standard: PSR12

**Additional context**
phpcbf can not fix because of some internal exception

**Comment by gsherwood** [comment]
This turned out to be the same issue as #3060, for which the fix will be in 3.5.7.
